notebooks/scrf.ipynb

```diff
@@ -0,0 +1,828 @@
+    "        coords=[\n",
+    "            np.linspace(start=-20000000, stop=20000000, num=360),\n",
+    "            np.linspace(start=-14000000, stop=14000000, num=181),\n",
+    "        ],\n",
+    "    )\n",
+    "    mill_template = mill_template.rio.write_crs(\"+proj=mill\")\n",
+    "\n",
+    "    projected = template.rio.write_crs(\"EPSG:4326\").rio.reproject_match(\n",
+    "        mill_template\n",
+    "    )\n",
+    "    # transform from m to km\n",
+    "    x = np.round(projected.x.values / 1000.0, 2)\n",
+    "    y = np.round(projected.y.values / 1000.0, 2)\n",
+    "\n",
+    "    # get number of days in the period\n",
+    "    t = pd.date_range(\n",
+    "        start=f\"{start_year}-01-01\", end=f\"{end_year}-12-31\", freq=\"D\"\n",
+    "    )\n",
+    "    print(len(x), len(y), len(t))\n",
+    "\n",
+    "    print(\"generating field\")\n",
+    "    # generate field\n",
+    "    time = np.arange(len(t))\n",
+    "    field = srf.structured((x, y, time))\n",
+    "\n",
+    "    print(\"making dataarray\")\n",
+    "    field = xr.DataArray(\n",
+    "        field,\n",
+    "        dims=[\"x\", \"y\", \"time\"],\n",
+    "        coords=[projected.x, projected.y, t],\n",
+    "    ).rio.write_crs(\"+proj=mill\")\n",
+    "\n",
+    "    # reprojecting\n",
+    "    print(\"reprojecting\")\n",
+    "    field = field.transpose(\"time\", \"y\", \"x\").rio.reproject(\"EPSG:4326\")\n",
+    "    field = field.rename({\"x\": \"lon\", \"y\": \"lat\"})\n",
+    "    template = template.rename({\"x\": \"lon\", \"y\": \"lat\"})\n",
+    "\n",
+    "    # regridding\n",
+    "    print(\"regridding\")\n",
+    "    regridder = xe.Regridder(\n",
+    "        field, template, \"bilinear\", extrap_method=\"nearest_s2d\"\n",
+    "    )\n",
+    "    regridded_field = regridder(field)\n",
+    "\n",
+    "    print(\"saving\")\n",
+    "    # save to file\n",
+    "    if variable == \"temp\":\n",
+    "        for v in [\"tasmax\", \"tasmin\"]:\n",
+    "            path = make_scrf_path(\"ERA5\", v, start_year, end_year)\n",
+    "            store = fsspec.get_mapper(\"az://flow-outputs/intermediate/\" + path)\n",
+    "            regridded_field.to_dataset(name=\"scrf\").to_zarr(\n",
+    "                store, mode=\"w\", consolidated=True\n",
+    "            )\n",
+    "    else:\n",
+    "        path = make_scrf_path(\"ERA5\", \"pr\", start_year, end_year)\n",
+    "        store = fsspec.get_mapper(\"az://flow-outputs/intermediate/\" + path)\n",
+    "        regridded_field.to_dataset(name=\"scrf\").to_zarr(\n",
+    "            store, mode=\"w\", consolidated=True\n",
+    "        )\n",
+    "\n",
+    "    return (\"processed\", path)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9e61ad51-deb2-4e76-9130-8ccf62cf3586",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "for variable in [\"temp\", \"pr\"]:\n",
```

Comment:
i feel like all of my comments about temp v. tmax v. tmin might be unnecessary.... but why is it sometimes `temp` and sometimes `tmax`/`tmin`?

Reply:
OH i just read the docs at the top again and i understand now. nevermind.